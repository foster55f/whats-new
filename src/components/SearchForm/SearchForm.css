header {
    border-bottom: black solid 2px;
    display: flex;
    justify-content: space-between;
    padding: 50px 0;
}

h1 {
    background-color: yellow;
    color:black;
    font-size: 50px;
    margin-left: 60px;
    padding: 0 15px 5px 22px;
}

input{
    width: 50%;
}

input {
    height: 50px;
    padding:10px;
    width: 50%;
    margin-top: 1%;
}

.searchButton {
    width:50%;
    height: 51px;
}

button {
    color: black;
    cursor: pointer;
    font-weight: 500px;
    background-color: yellow;
    margin:5px;
    padding: 5px;
    cursor: pointer;
    border-radius: 5px;
    width:45%;
}

.searchContainer{
    display: flex;
    width: 45%; 
    margin: auto;
    margin-left: 25%
}