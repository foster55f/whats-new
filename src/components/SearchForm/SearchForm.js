import React, { Component } from 'react';
import './SearchForm.css';

class SearchForm extends Component {
    constructor() {
      super();
      this.state = {
        
        }   
    }
    render() {
        return (
          <header>
            <input
              type='text'
              placeholder='Title'
              name='title'
              value={this.state.title}
              onChange={event => this.handleChange(event)}
            />
            <button onClick={event => this.submitIdea(event)}>Search</button>
          </header>
        )
      }
    
    }

export default SearchForm;