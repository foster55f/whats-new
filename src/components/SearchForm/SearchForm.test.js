import React from 'react';
import { shallow } from 'enzyme';
import SearchForm from './SearchForm';

describe('SearchForm', () => {
    it('should match the snapshot', () => {
        const filterSearchMock = jest.fn()
        const wrapper = shallow(<SearchForm search={filterSearchMock} />)
        
        expect(wrapper).toMatchSnapshot();
    })

    it('should update state when search is inputted', () => {
        const filterSearchMock = jest.fn()
        const wrapper = shallow(<SearchForm search={filterSearchMock} />)

        const searchTime = {
            target: {
                value: 'spider-man'
            }
        }
        
        wrapper.instance().handleChange(searchTime);
        expect(wrapper.state('searchField')).toEqual(searchTime.target.value)
    })

    it('should update state when search is inputted', () => {
        const filterSearchMock = jest.fn()
        const wrapper = shallow(<SearchForm search={filterSearchMock} />)

        const searchTime = {
            target: {
                value: 'spider-man'
            }
        }
        
        wrapper.instance().handleChange(searchTime);
        expect(wrapper.state('searchField')).toEqual(searchTime.target.value)
    })
})