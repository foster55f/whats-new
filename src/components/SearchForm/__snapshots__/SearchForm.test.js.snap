// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should match the snapshot 1`] = `
<header>
  <h1>
    What's New
  </h1>
  <div
    className="searchContainer"
  >
    <input
      name="title"
      onChange={[Function]}
      placeholder="Search"
      type="text"
      value=""
    />
    <button
      className="searchButton"
      onClick={[Function]}
    >
      Search
    </button>
  </div>
</header>
`;
